### PR TITLE
[OCB] Copy+Update CWAgent's `awsapplicationsignals` into the `processor/` directory

### PR DESCRIPTION
**Pull request by jj22ee**
**Description:** <Describe what has changed.>

This PR is to create the initial baseline commit of the Application Signals Processor that future commits will modify in order to properly make this processor an externally usable Go module

This PR does 2 things
1. Copy the following directory with 0 changes into the `processor` directory - https://github.com/aws/amazon-cloudwatch-agent/tree/main/plugins/processors/awsapplicationsignals
2. Rename copied directory from `awsapplicationsignals` to `awsapplicationsignalsprocessor`

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

N/A. As this PR is just a direct copy + rename of the appsignals processor, nothing is expected to work.

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by mxiamxia — APPROVED** [review]
Approved PR as we're just copying the code over.

**Comment by jj22ee** [comment]
The first 2 commits are for the direct copy of the Application Signals Processor.

In the 3rd commit, additional changes are made:
1. Update all imports of `github.com/aws/amazon-cloudwatch-agent/plugins/processors/awsapplicationsignals` to `github.com/amazon-contributing/opentelemetry-collector-contrib/processor/awsapplicationsignalsprocessor`
2. Add Makefile
3. Add `processor/awsapplicationsignalsprocessor/go.mod` and `go.sum`
4. Copy over `ecsutil.go` [from CWAgent](https://github.com/aws/amazon-cloudwatch-agent/blob/main/translator/util/ecsutil/ecsutil.go)
    - Remove dependency on `github.com/aws/amazon-cloudwatch-agent/translator/config` by copying the `config.RUN_IN_CONTAINER` and `config.RUN_IN_CONTAINER_TRUE` values
6. Copy over `httpclient.go` [from CWAgent](https://github.com/aws/amazon-cloudwatch-agent/blob/main/translator/util/httpclient/httpclient.go)
7. Update `processor/awsapplicationsignalsprocessor/internal/normalizer/attributesnormalizer.go`
    - Change attribute value of `common.MetricAttributeTelemetryAgent` from `"CWAgent/%s", version.Number()` to `opentelemetry-collector/%s", GetCollectorVersion()`.
    - Add ` processor/awsapplicationsignalsprocessor/internal/normalizer/version_info.go` to get the Collector Version via `GetCollectorVersion()`
8. Remove `eksdetector` dependency in `processor/awsapplicationsignalsprocessor/internal/resolver/kubernetes_test.go`

File: processor/awsapplicationsignalsprocessor/README.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+           - dimension: Operation
+             match: "*"
+        replacements:
+          - target_dimension: RemoteOperation
+            value: "This is a test string"
+        action: replace
+        rule_name: "replace01"
+```
+
+## Amazon CloudWatch Agent Configuration Example
```
Comment (jefchien) [inline processor/awsapplicationsignalsprocessor/README.md]: Do we still want this in here? If the idea is to separate the processor from the agent, then I don't think the agent configuration should be part of the README.

**Review by jefchien — APPROVED** [review]

**Comment by jj22ee** [comment]
The `check-links` workflow is failing due to new `markdown-link-check` version. It is unrelated to this PR.
It was fixed in upstream via https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36231

**Pull request merged**